File: php/testcode/deserial_011.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_deser_phar_wrapper
function deserial_phar_wrapper(BenchmarkRequest $req): BenchmarkResponse {
    $file = $req->param('file');
    $content = file_get_contents("phar://" . $file); // vuln-code-snippet vuln-line php_deser_phar_wrapper
    return BenchmarkResponse::ok($content);
}
// vuln-code-snippet end php_deser_phar_wrapper

File: ruby/testcode/deserial_011.rb
require_relative 'shared'
require 'base64'

# vuln-code-snippet start ruby_deser_marshal_base64
def resume_session(req)
  encoded = req.post('session')
  obj = Marshal.load(Base64.decode64(encoded)) # vuln-code-snippet vuln-line ruby_deser_marshal_base64
  user_id = obj[:user_id].to_s
  BenchmarkResponse.ok("session for user #{user_id}")
end
# vuln-code-snippet end ruby_deser_marshal_base64
